### PR TITLE
Remove $NUPIC 2.0

### LINKED ISSUE #1947: Swarming returns JSON error when NUPIC environment variable is unset
To replicate:

1. install nupic from source following README instructions
1. `cd examples/opf/clients/hotgym/prediction/one_gym`
1. `unset NUPIC` (if set)
1. `python swarm.py`

Error:

```
∙ ./swarm.py
This script runs a swarm on the input data (rec-center-hourly.csv) and
creates a model parameters file in the `model_params` directory containing
the best model found by the swarm. Dumps a bunch of crud to stdout because
that is just what swarming does at this point. You really don't need to
pay any attention to it.

=================================================
= Swarming on rec-center-hourly data...
= Medium swarm. Sit back and relax, this could take awhile.
=================================================
Generating experiment files in directory: /Users/mtaylor/nta/nupic/examples/opf/clients/hotgym/prediction/one_gym/swarm...
Writing  313 lines...
Writing  113 lines...
done.
None
Successfully submitted new HyperSearch job, jobID=1411
Evaluated 0 models
HyperSearch finished!
Worker completion message: None

Results from all experiments:
----------------------------------------------------------------
Generating experiment files in directory: /var/folders/c1/fc9tt93s2p5b9d3wp185qxwc0000gn/T/tmp6234aw...
Writing  313 lines...
Writing  113 lines...
done.
None
json.loads(jobInfo.results) raised an exception.  Here is some info to help with debugging:
jobInfo:  _jobInfoNamedTuple(jobId=1411, client=u'GRP', clientInfo=u'', clientKey=u'', cmdLine=u'$HYPERSEARCH', params=u'{"hsVersion": "v2", "maxModels": null, "persistentJobGUID": "0d6edaa1-d262-11e4-94c6-600308a99116", "useTerminators": false, "description": {"includedFields": [{"fieldName": "timestamp", "fieldType": "datetime"}, {"maxValue": 53.0, "fieldName": "kw_energy_consumption", "fieldType": "float", "minValue": 0.0}], "streamDef": {"info": "kw_energy_consumption", "version": 1, "streams": [{"info": "Rec Center", "source": "file://rec-center-hourly.csv", "columns": ["*"]}]}, "inferenceType": "TemporalMultiStep", "inferenceArgs": {"predictionSteps": [1], "predictedField": "kw_energy_consumption"}, "iterationCount": -1, "swarmSize": "medium"}}', jobHash='\ro\x9b\xd4\xd2b\x11\xe4\xa0\xfa`\x03\x08\xa9\x91\x16', status=u'notStarted', completionReason=None, completionMsg=None, workerCompletionReason=u'success', workerCompletionMsg=None, cancel=0, startTime=None, endTime=None, results=None, engJobType=u'hypersearch', minimumWorkers=1, maximumWorkers=4, priority=0, engAllocateNewWorkers=1, engUntendedDeadWorkers=0, numFailedWorkers=0, lastFailedWorkerErrorMsg=None, engCleaningStatus=u'notdone', genBaseDescription=None, genPermutations=None, engLastUpdateTime=datetime.datetime(2015, 3, 24, 20, 12, 10), engCjmConnId=None, engWorkerState=None, engStatus=None, engModelMilestones=None)
jobInfo.results:  None
EXCEPTION:  expected string or buffer
Traceback (most recent call last):
  File "./swarm.py", line 109, in <module>
    swarm(INPUT_FILE)
  File "./swarm.py", line 101, in swarm
    modelParams = swarmForBestModelParams(SWARM_DESCRIPTION, name)
  File "./swarm.py", line 78, in swarmForBestModelParams
    verbosity=0
  File "/Users/mtaylor/nta/nupic/nupic/swarming/permutations_runner.py", line 276, in runWithConfig
    return _runAction(runOptions)
  File "/Users/mtaylor/nta/nupic/nupic/swarming/permutations_runner.py", line 217, in _runAction
    returnValue = _runHyperSearch(runOptions)
  File "/Users/mtaylor/nta/nupic/nupic/swarming/permutations_runner.py", line 160, in _runHyperSearch
    metricsKeys=search.getDiscoveredMetricsKeys())
  File "/Users/mtaylor/nta/nupic/nupic/swarming/permutations_runner.py", line 825, in generateReport
    results = json.loads(jobInfo.results)
  File "/Users/mtaylor/nta/nupic/nupic/support/object_json.py", line 163, in loads
    json.loads(s, object_hook=objectDecoderHook, **kwargs))
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/__init__.py", line 351, in loads
    return cls(encoding=encoding, **kw).decode(s)
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/json/decoder.py", line 365, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
TypeError: expected string or buffer
```

CC @pehlert

### PR DESCRIPTION
**Pull request by passiweinberger**
Sorry, I deleted the branch from the previous PR... 

fixes #1947

### CODE REVIEW

**Comment by david-ragazzi** [comment]
@passiweinberger From I understood this new PR replaces both https://github.com/numenta/nupic/pull/1970 and https://github.com/numenta/nupic/pull/1968, right?

**Comment by david-ragazzi** [comment]
I liked these changes. It make simplify and unify the use of nupic root.

File: nupic/support/__init__.py [hunk 1]
```diff
@@ -402,6 +403,10 @@ def initLogging(verbose=False, console='stdout', consoleLevel='DEBUG'):
   #   module
   configFilename = 'nupic-logging.conf'
   configFilePath = resource_filename("nupic.support", configFilename)
+  
+  # if 'NTA_LOG_DIR' not in os.environ:
+  #  os.environ['NTA_LOG_DIR'] = os.path.join(NUPIC_ROOT, '../logs')
```
Comment (david-ragazzi) [inline nupic/support/__init__.py]: > As #2005 should have fixed it here...
> Left the chnge commented

In this case, you can remove these lines.

**Comment by david-ragazzi** [comment]
:+1: (with condition of you fix what I said).

If @scottpurdy and others doesn't have any questions, we can merge it.

File: nupic/__init__.py [hunk 2]
```diff
@@ -19,3 +19,5 @@
 # http://numenta.org/licenses/
 # ----------------------------------------------------------------------
 __version__ = "0.2.4.dev0"
+
+NUPIC_ROOT = os.environ.get('NUPIC', os.path.dirname(os.path.realpath(__file__)))
```
Comment (rhyolight) [inline nupic/__init__.py]: You need to `import os` in this file. See failing build jobs: https://travis-ci.org/numenta/nupic/builds/66349699

**Comment by rhyolight** [comment]
Aside from failing builds (and suggested fix), LGTM. @scottpurdy ?

File: nupic/__init__.py (hunk 2 above)
Comment (passiweinberger) [inline nupic/__init__.py]: Ups :S

**Comment by david-ragazzi** [comment]
> command = ['python', os.path.join(testDir, 'experiments', 'classification', 'makeDatasets.py')]
> retval = call(command)
>  self.assertEqual(retval, 0)

> https://travis-ci.org/numenta/nupic/jobs/66414407#L4021

@passiweinberger Travis failed. Print the output text of `call()` to check what was the problem with the nupic path.

**Comment by scottpurdy** [comment]
This is great to get $NUPIC out of the codebase. Can pkg_resources be used in these cases instead of hard coding the root and doing everything relative to that?

**Comment by passiweinberger** [comment]
Yep, pkg-resourses look like a good fit for that. I'll go with that one if
prefered.

**Comment by passiweinberger** [comment]
Oh but with pkg-resourses we were still 'hardcoding' the main root and the
work relative from that, wouldn't we?
https://pythonhosted.org/setuptools/pkg_resources.html
in declare_namespace ()

**Comment by scottpurdy** [comment]
With pkg_resources you specify a module and a path relative to the module. So there is nothing hard-coded. In one place you can have this:

```
from nupic import datafiles

HOTGYM = pkg_resources.resource_string(datafiles, "extra/.../hotgym.csv")
```

And in another you can look up a file relative to a different module.

**Comment by passiweinberger** [comment]
And David, I'll address that tmw. But why is L4017 even there? L4008
already sets the testDir to opfDir anyways? Looks a bit like the if not
statement is obsolete :$

**Comment by passiweinberger** [comment]
Ok I'll go change it all to use pgk-resources then:)

**Comment by passiweinberger** [comment]
@david-ragazzi the issue is that i cannot call python on relative paths. 
pkg-resourses will fix that i think

File: tests/integration/nupic/opf/opf_experiment_results_test.py [hunk 3]
```diff
@@ -55,7 +56,7 @@ def testExperimentResults(self):
     examples/prediction directory in the install tree (same as predictionDir)
 
     """
-    nupicDir = os.environ['NUPIC']
+    nupicDir = os.path.join(NUPIC_ROOT, "..")
```
Comment (passiweinberger) [inline tests/integration/nupic/opf/opf_experiment_results_test.py]: which module would you specify here ? neither examples nor opf are importable....

Comment (passiweinberger) [inline tests/integration/nupic/opf/opf_experiment_results_test.py]: or just pkg_resources.resource_string(nupic)

**Comment by passiweinberger** [comment]
@scottpurdy that doesn't work out like this... 

```
>>> HOTGYM = pkg_resources.resource_string(datafiles, "extra/.../hotgym.csv")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/pascal/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 1172, in resource_string
    return get_provider(package_or_requirement).get_resource_string(
  File "/home/pascal/anaconda/lib/python2.7/site-packages/pkg_resources/__init__.py", line 435, in get_provider
    __import__(moduleOrReq)
TypeError: __import__() argument 1 must be string, not module
```

and when you try it as string: 
you get an IOError: No such file or directory...

File: nupic/support/__init__.py [hunk 4]
```diff
@@ -83,6 +83,7 @@
 
 from configuration import Configuration
 from nupic.support.fshelpers import makeDirectoryFromAbsolutePath
+from nupic import NUPIC_ROOT
```
Comment (breznak) [inline nupic/support/__init__.py]: the variable seems not to be used anywhere here

**Comment by scottpurdy** [comment]
You can try passing "nupic.datafiles" instead of the imported module.

**Comment by passiweinberger** [comment]
in nupic/init the NUPIC_ROOT is lef, as it might be useful for other purposes, is that good? :)

**Comment by scottpurdy** [comment]
> in nupic/init the NUPIC_ROOT is lef, as it might be useful for other purposes, is that good? :)

Probably better not to have it since people may use it instead of pkg_resources if they see it there.

> This is weired :/ in my console it returns 0 .. :S

Try using pkg_resources instead of `NUPIC_ROOT`

**Comment by passiweinberger** [comment]
I did, changed it back to see if anything works :s Both give error 2 file
not found :s

**Comment by scottpurdy** [comment]
What is the file path in the repo?

**Comment by david-ragazzi** [comment]
A last question: Why did you discard `os.path.join()` function? We can have problems when work Windows.

Apart this, this PR is really great!

File: examples/opf/experiments/multistep/base/permutations_simple_3.py [hunk 5]
```diff
@@ -58,8 +58,7 @@
   'predictedField': 'field2',
   'predictionSteps': [1,3],
   
-  'dataSource': 'file://%s' % (os.path.join(os.environ['NUPIC'], 'examples', 'opf', 
-                    'experiments', 'multistep', 'datasets', 'simple_3.csv')),
+  'dataSource': 'file://%s' % (resource_filename("nupic","examples/opf/experiments/multistep/datasets/simple_3.csv")),
```
Comment (david-ragazzi) [inline examples/opf/experiments/multistep/base/permutations_simple_3.py]: Why did you removed `os.path.join`?

File: tests/integration/nupic/opf/expgenerator_test.py [hunk 6]
```diff
@@ -118,7 +118,7 @@ def setUp(self):
     global g_myEnv
     if not g_myEnv:
       # Setup environment
-      params = type('obj', (object,), {'installDir' : os.environ['NUPIC']})
+      params = type('obj', (object,), {'installDir' : resource_filename("nupic","")})
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/expgenerator_test.py]: Please leave 1 space char between colons here and other places in this PR.

File: tests/unit/nupic/support/configuration_test.py [hunk 7]
```diff
@@ -519,8 +520,7 @@ def testGetConfigPathsForNone(
     configuration.Configuration._configPaths = None  # pylint: disable=W0212
     result = configuration.Configuration.getConfigPaths()
     self.assertTrue(isinstance(result, list))
-    self.assertEqual(result, [os.path.join(os.environ['NUPIC'],
-                                           'config', 'default')])
+    self.assertEqual(result, [resource_filename("nupic","config/default")])
```
Comment (david-ragazzi) [inline tests/unit/nupic/support/configuration_test.py]: Space between comma.

File: tests/unit/nupic/support/custom_configuration_test.py [hunk 8]
```diff
@@ -803,8 +803,7 @@ def testGetConfigPathsForNone(
     configuration.Configuration._configPaths = None  # pylint: disable=W0212
     result = configuration.Configuration.getConfigPaths()
     self.assertTrue(isinstance(result, list))
-    self.assertEqual(result, [os.path.join(os.environ['NUPIC'],
-                                           'config', 'default')])
+    self.assertEqual(result, [resource_filename("nupic","config/default")])
```
Comment (david-ragazzi) [inline tests/unit/nupic/support/custom_configuration_test.py]: Space between comma.

File: tests/swarming/nupic/swarming/swarming_test.py [hunk 9]
```diff
@@ -255,8 +257,8 @@ def _generateHSJobParams(self,
 
       # Form the stream definition
       if dataPath is None:
-        dataPath = os.path.join(os.environ['NUPIC'], 'examples', 'prediction',
-                        'data', 'extra', 'qa', "hotgym", "qa_hotgym.csv")
+        dataPath = resource_filename("nupic.data", "extra/qa/hotgym/qa_hotgym.csv")
```
Comment (david-ragazzi) [inline tests/swarming/nupic/swarming/swarming_test.py]: Removal of `os.path.join`.

File: tests/integration/nupic/opf/opf_experiments_test.py [hunk 10]
```diff
@@ -33,7 +34,7 @@
 
 # Globals
 EXCLUDED_EXPERIMENTS = [] # none for now
-PREDICTION_DIR = os.path.join(os.environ['NUPIC'], 'examples', 'opf')
+PREDICTION_DIR = resource_filename("nupic", "../examples/opf")
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/opf_experiments_test.py]: Removal of `os.path.join`.

File: tests/integration/nupic/opf/opf_experiment_results_test.py [hunk 11]
```diff
@@ -95,7 +96,7 @@ def testExperimentResults(self):
     # Run from the test directory so that we can find our experiments
     os.chdir(testDir)
 
-    runExperiment = os.path.join(nupicDir, 'scripts',  'run_opf_experiment.py')
+    runExperiment = resource_filename("nupic", "../scripts/run_opf_experiment.py")
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/opf_experiment_results_test.py]: Removal of `os.path.join`.

File: tests/integration/nupic/opf/opf_experiment_results_test.py [hunk 12]
```diff
@@ -55,9 +57,8 @@ def testExperimentResults(self):
     examples/prediction directory in the install tree (same as predictionDir)
 
     """
-    nupicDir = os.environ['NUPIC']
 
-    opfDir = os.path.join(nupicDir, "examples", "opf")
+    opfDir = resource_filename("nupic", "../examples/opf")
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/opf_experiment_results_test.py]: Removal of `os.path.join`.

File: tests/integration/nupic/opf/opf_description_template_test/opf_description_template_test.py [hunk 13]
```diff
@@ -58,7 +58,7 @@ def __init__(self):
 
     # This is where we find OPF binaries (e.g., run_opf_experiment.py, etc.)
     # In the autobuild, it is a read-only directory
-    self.__opfBinDir = os.path.join(nupicDir, "scripts")
+    self.__opfBinDir = resource_filename("nupic", "../scripts")
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/opf_description_template_test/opf_description_template_test.py]: Removal of `os.path.join`.

File: tests/integration/nupic/opf/opf_description_template_test/opf_description_template_test.py [hunk 14]
```diff
@@ -47,9 +49,7 @@ class MyTestEnvironment(object):
 
   def __init__(self):
 
-    nupicDir = os.environ['NUPIC']
-
-    examplesDir = os.path.join(nupicDir, "examples")
+    examplesDir = resource_filename("nupic", "../examples")
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/opf_description_template_test/opf_description_template_test.py]: Removal of `os.path.join`.

File: tests/integration/nupic/opf/expgenerator_test.py [hunk 15]
```diff
@@ -1947,7 +1947,7 @@ def _getTestList():
   # Our custom options (that don't get passed to unittest):
   customOptions = ['--installDir', '--verbosity', '--logLevel']
   parser.add_option(
-    "--installDir", dest="installDir", default=os.environ['NUPIC'],
+    "--installDir", dest="installDir", default=resource_filename("nupic",""),
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/expgenerator_test.py]: No space between comma.

**Comment by passiweinberger** [comment]
Because resource_filname joins them itself... So shall I switch it back?

**Comment by david-ragazzi** [comment]
> Because resource_filname joins them itself... So shall I switch it back? 

While we don't have a decision about, let's leave as is.

**Comment by passiweinberger** [comment]
Like that?

**Comment by david-ragazzi** [comment]
:+1: If @scottpurdy doesn't have more questions, I'll merge it.

**Comment by passiweinberger** [comment]
Wait for travis though... I'm not sure about os join to work ...

File: tests/unit/nupic/support/configuration_test.py [hunk 16]
```diff
@@ -519,8 +520,7 @@ def testGetConfigPathsForNone(
     configuration.Configuration._configPaths = None  # pylint: disable=W0212
     result = configuration.Configuration.getConfigPaths()
     self.assertTrue(isinstance(result, list))
-    self.assertEqual(result, [os.path.join(os.environ['NUPIC'],
-                                           'config', 'default')])
+    self.assertEqual(result, [resource_filename("nupic", "config/default")])
```
Comment (david-ragazzi) [inline tests/unit/nupic/support/configuration_test.py]: Here too.

File: tests/unit/nupic/support/custom_configuration_test.py [hunk 17]
```diff
@@ -803,8 +803,7 @@ def testGetConfigPathsForNone(
     configuration.Configuration._configPaths = None  # pylint: disable=W0212
     result = configuration.Configuration.getConfigPaths()
     self.assertTrue(isinstance(result, list))
-    self.assertEqual(result, [os.path.join(os.environ['NUPIC'],
-                                           'config', 'default')])
+    self.assertEqual(result, [resource_filename("nupic", "config/default")])
```
Comment (david-ragazzi) [inline tests/unit/nupic/support/custom_configuration_test.py]: Here too.

**Comment by david-ragazzi** [comment]
You did forget update  two lines.

**Comment by passiweinberger** [comment]
Know why Win may fail there? Looks way cleaner w/o though...

**Comment by david-ragazzi** [comment]
> Know why Win may fail there? Looks way cleaner w/o though...

Probably Win not fail with resource_package but this could make the entire code base inconsistent with the convention, where we use os.path.join to avoid use `/` which sometimes is not accepted by Windows.

File: examples/opf/experiments/multistep/base/permutations_simple_3.py [hunk 18]
```diff
@@ -58,8 +59,7 @@
   'predictedField': 'field2',
   'predictionSteps': [1,3],
   
-  'dataSource': 'file://%s' % (os.path.join(os.environ['NUPIC'], 'examples', 'opf', 
-                    'experiments', 'multistep', 'datasets', 'simple_3.csv')),
+  'dataSource': 'file://%s' % (os.path.join(resource_filename("nupic", ""), "examples", "opf", "experiments", "multistep", "datasets", "simple_3.csv")),
```
Comment (scottpurdy) [inline examples/opf/experiments/multistep/base/permutations_simple_3.py]: need to break this line to stay under 80 characters

Comment (scottpurdy) [inline examples/opf/experiments/multistep/base/permutations_simple_3.py]: And you should do the os.path.join inside the resource_filename call

File: tests/unit/nupic/support/configuration_test.py [hunk 19]
```diff
@@ -519,8 +520,7 @@ def testGetConfigPathsForNone(
     configuration.Configuration._configPaths = None  # pylint: disable=W0212
     result = configuration.Configuration.getConfigPaths()
     self.assertTrue(isinstance(result, list))
-    self.assertEqual(result, [os.path.join(os.environ['NUPIC'],
-                                           'config', 'default')])
+    self.assertEqual(result, [os.path.join(resource_filename("nupic", ""), "config", "default")])
```
Comment (scottpurdy) [inline tests/unit/nupic/support/configuration_test.py]: `assertListEqual`

**Comment by scottpurdy** [comment]
Two things that you should pay attention to throughout this PR: 80 character line limit, and using `os.path.join` first, then `resource_filename` on the result. In fact, there are many cases where I saw `resource_filename("nupic", "")` which should pretty much never be done since the whole point is to look up a filename but you give no path.

**Comment by scottpurdy** [comment]
@david-ragazzi - Are you sure that there is a problem with something like this: `resource_filename("nupic", "some/path")`

I would think pkg_resources would handle that fine on different platforms.

**Comment by david-ragazzi** [comment]
>  Are you sure that there is a problem with something like this: 

Not a real problem, but it's because this brings inconsistency. It's not a blocking issue, if @passiweinberger wants revert it, not problem to me.

**Comment by david-ragazzi** [comment]
BTW once @passiweinberger fix what you propose, I'll merge, ok?

**Comment by passiweinberger** [comment]
I think w/o path.join it looks way cleaner, but I do not have tested it on Win, so if there might occur a problem with '/', we should leave the join to be sure...

**Comment by scottpurdy** [comment]
indentation for the line breaks is off. You can take a look at [PEP-8](https://www.python.org/dev/peps/pep-0008/#indentation) or use Pylint to see how the breaks should be formatted.

File: examples/opf/experiments/multistep/base/permutations_simple_3.py [hunk 20]
```diff
@@ -58,8 +59,8 @@
   'predictedField': 'field2',
   'predictionSteps': [1,3],
   
-  'dataSource': 'file://%s' % (os.path.join(os.environ['NUPIC'], 'examples', 'opf', 
-                    'experiments', 'multistep', 'datasets', 'simple_3.csv')),
+  'dataSource': 'file://%s' % (resource_filename("nupic", 
+              os.path.join("examples", "opf", "experiments", "multistep", "datasets", "simple_3.csv"))),
```
Comment (david-ragazzi) [inline examples/opf/experiments/multistep/base/permutations_simple_3.py]: Change to something like this:

```
  relativePath = os.path.join("examples", "opf", "experiments", "multistep",
                              "datasets", "simple_3.csv")
  'dataSource': 'file://%s' % (resource_filename("nupic", relativePath)),
```

**Comment by david-ragazzi** [comment]
> indentation for the line breaks is off. You can take a look at PEP-8 or use Pylint to see how the breaks should be formatted.

@passiweinberger Scott meant line breaks for function parameters which next parameters should be indentend with the first parameter. I suggested some changes to ease this update. Just do it and I merge.

File: tests/integration/nupic/opf/expgenerator_test.py [hunk 21]
```diff
@@ -49,7 +49,7 @@
                                            InferenceElement)
 
 LOGGER = logging.getLogger(__name__)
-HOTGYM_INPUT = resource_filename("nupic.datafiles", "extra/hotgym/hotgym.csv")
+HOTGYM_INPUT = resource_filename("nupic.datafiles", os.path.join("extra", "hotgym", "hotgym.csv"))
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/expgenerator_test.py]: Adjust line break:

```
HOTGYM_INPUT = resource_filename("nupic.datafiles",
                                  os.path.join("extra", "hotgym", "hotgym.csv"))

```

File: tests/integration/nupic/opf/expgenerator_test.py [hunk 22]
```diff
@@ -1947,7 +1947,7 @@ def _getTestList():
   # Our custom options (that don't get passed to unittest):
   customOptions = ['--installDir', '--verbosity', '--logLevel']
   parser.add_option(
-    "--installDir", dest="installDir", default=os.environ['NUPIC'],
+    "--installDir", dest="installDir", default=resource_filename("nupic", ""),
```
Comment (david-ragazzi) [inline tests/integration/nupic/opf/expgenerator_test.py]: Adjust line break:

```
  parser.add_option("--installDir", dest="installDir",
        default=resource_filename("nupic", ""),
        help="Path to the NTA install directory [default: %default].")
```

File: tests/swarming/nupic/swarming/swarming_test.py [hunk 23]
```diff
@@ -255,8 +257,9 @@ def _generateHSJobParams(self,
 
       # Form the stream definition
       if dataPath is None:
-        dataPath = os.path.join(os.environ['NUPIC'], 'examples', 'prediction',
-                        'data', 'extra', 'qa', "hotgym", "qa_hotgym.csv")
+        dataPath = resource_filename("nupic.data", os.path.join("extra",
```
Comment (david-ragazzi) [inline tests/swarming/nupic/swarming/swarming_test.py]:
```
        dataPath = resource_filename("nupic.data",
                                     os.path.join("extra", "qa", "hotgym",
                                                  "qa_hotgym.csv"))
```

**Comment by passiweinberger** [comment]
Ahh ups :S

**Comment by david-ragazzi** [comment]
@rhyolight Maybe the documentation and Travis scripts needs be updated.

**Pull request merged**